### DISCUSSION TITLE
CV: ChemSpectra does not process PalmSens PSTrace pssession files from inbox

### DISCUSSION THREAD

**Discussion by tilfischer** [ComPlat/chemotion_ELN#2383] [Q&A] [answered]
- I am an ELNer admin and user

# Expected Behavior

PalmSens PSTrace pssession file is dropped from the inbox to an analysis and processed by chemspectra after saving the sample.

# Current Behavior

PalmSens PSTrace psssession file is dropped from the inbox to an analysis and not processed by ChemSpectra after saving the sample.

# Failure Information 

PalmSens PSTrace pssession files can be uploaded to an analysis manually and is processed by ChemSpectra, while the file added via drag-and-drop from the inbox to the analysis is not processed. We could also drag-and-drop a pssession file from the inbox to the analysis and save this (no processing), download the pssession in the pop-up modal and reupload it manually to another analysis and the pssession got processed and we could continue with analysis.

## Steps to Reproduce

Drop pssession file to folder watched by filewatcher. Wait until filewatcher presents the pssession file in the inbox. Drag-and-drop it to an analysis of an sample and save the sample. The pssession file is not converted to JCAMP-DX and data cannot be analysed.

## Context

 - OS: WebServer running OpenSuse Tumbleweed, Clients Windows 10 or Windows 11 or macOS with latest Chrome browser.
 - Version: 1.8.2, Base revision: d5cac4c27 with paggyhuang/converter:1.2.0

Best,
Tillmann

**Comment by tilfischer** [comment]
Dear all,

more insights and tests:

When a pssession file was transferred to the folder watched by datacollector, this actually gives two files in the inbox of the user (not mentioned above, sorry for that): 1) the pssession and 2) a ZIP containing some JCAMP-DX files. 

Drag-and-drop of both files in the inbox to an analysis causes ChemSpectra to process the data and the user can continue with analysis. We initially ignored the ZIP, as it was unclear to us where this ZIP comes from. That both need to be added to an analysis via drag-and-drop needs to be added to the documentation. 

I also tested to drop a zipped pssession file to the folder watched by the datacollector, assuming that Chemotion might unzip this, add the data in the additional zip, mentioned in the paragraph above, and zipps all again to one ZIP and displays this ZIP in the inbox. However, this was not the case. The inbox only showed the zipped pssession ZIP but not the additional ZIP. Drag-and-drop of the zipped pssession ZIP to an analyses did not lead to processing of the pssession ChemSpectra.

This issue can be closed as soon as drag-and-drop of both files in the inbox (CV data file and ZIP) is mentioned in the documentation.

Best,
Tillmann

**Comment by herrdivad** [comment]
Hi @tilfischer,
As you have already recognized yourself "pssession" upload is currenty supported from Hard Drive while ".zip" (e.g. special ones from the converter) is supported from INBOX (as far as I know, additional pssession from INBOX could be added but is not needed). I myself already mentioned it to the team because it is the case across all (or most) devices and data types.
As far as I know, the current behaviour was intended from the developers but we can open a discussion about it.
Also unzipping for user zip files is currently also not supported.
About documentation: I planned to mention this different behavior also in the futur video.

**Comment by tilfischer** [comment]
Dear David,

we have also tested this. Drag and drop of only the ZIP is not sufficient. The PSSESSION file in the inbox also need to be added to the dataset of the analysis to allow for processing by ChemSpectra. We also downloaded the ZIP and checked its content. It does not include the PSSESSION file but some JCAMP-DX files.

We also tested not to drop the PSSESSION file to the folder watched by the filewatcher but a ZIP of the PSSESSION file. This led to this ZIP showing up in the INBOX, while the other ZIP with the JCAMP-DX files was not shown in the inbox. Dropping the ZIP and saving the sample did not lead to file conversion and user cannot proceed with analysis.

Best,
Tillmann

**Comment by tilfischer** [comment, marked as answer]
Here are test files: [Archiv.zip](https://github.com/ComPlat/chemotion_ELN/files/14773066/Archiv.zip)

The ZIP (I had to zip it to be able to upload it here) contains a PSSESSION and a ZIP. The PSSESSION was dropped in the local folder watched. The inbox then displayed this PSSESSION and the ZIP (in the ZIP). Both need to be added to a dataset of an analysis via drag-and-drop. I checked the ZIP in the ZIP is a bagit container with JCAMP-DX files in the data folder. 

Some conversion already takes place before the data is added to the dataset of an analysis via drag-and-drop and some more files are obtained by format conversion if the sample is saved. 

User simply should be aware of this i.e. that both need to be drag-and-dropped need to be added to the documenation.

Best,
Tillmann

**Comment by PiTrem** [comment]
Moving as discussion

I am adding some more warning in the docs

see  https://github.com/ComPlat/chemotion_saurus/pull/405 (not sure where-else to have this)
